controllers/operatorchecker/operatorchecker_controller.go

```diff
@@ -48,43 +52,87 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Re
 	if _, ok := subscriptionInstance.Labels[constant.OpreqLabel]; !ok {
 		return
 	}
-	if subscriptionInstance.Status.State != "" {
-		return
-	}
-	for _, condition := range subscriptionInstance.Status.Conditions {
-		if condition.Type == "ResolutionFailed" && condition.Reason == "ConstraintsNotSatisfiable" {
-			csvList := &olmv1alpha1.ClusterServiceVersionList{}
-			opts := []client.ListOption{
-				client.InNamespace(subscriptionInstance.Namespace),
-			}
-			if err := r.Client.List(ctx, csvList, opts...); err != nil {
-				return ctrl.Result{}, client.IgnoreNotFound(err)
+
+	if subscriptionInstance.Status.CurrentCSV == "" && subscriptionInstance.Status.State == "" {
+		// cover fresh install case
+		csv, err := r.getCSVBySubscription(ctx, subscriptionInstance)
+		if err != nil {
+			// not found csv
+			klog.Error(err)
+			return
+		}
+		err = wait.PollImmediate(time.Second*3, time.Minute*1, func() (bool, error) {
```

Comment:
The logic here should be like this:
In 1 minute if the status is ineligible, then re-queue for the next reconciliation. Otherwise, wait until 1 minute